mantidproject/mantid
# ILL loaders cleanup

**Description of work.**

Removed default values for:
`m_offsetTheta`, `m_pixelHeight` from `LoadILLDiffraction`, `m_numberOfTubes`, `m_numberOfPixelsPerTube`, `m_numberOfChannels`, `m_numberOfSimpleDetectors`, `m_firstTubeAngleRounded` from `LoadILLIndirect2`.

Moved the following currently constant values from loaders to relevant IPF:
`D20_NUMBER_DEAD_PIXELS`, `D20_PIXEL_SIZE`, `D2B_NUMBER_PIXELS_IN_TUBES`, `D7_NUMBER_BANKS`, `D7_NUMBER_PIXELS_BANK`.

**To test:**



1. LoadEmptyInstrument for D7, D2B and D20 to check if the parameters moved to the IPF are loaded for each instance of the instrument.

2. Load unit test data for IN16B, D7, D20, D2B, IN4, IN5, IN6, PANTHER. There should be no errors.

Fixes #29850 

This does not require release notes because these are internal changes in the implementation.



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
